packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1569,6 +1622,10 @@ export class MergeTree {
                 newSegment.seq = seq;
                 newSegment.localSeq = localSeq;
                 newSegment.clientId = clientId;
+                if (this.options?.trackAttribution) {
```

Comment:
should this be skipped if newSegment.Seq === Unassigned?

Reply:
Might as well avoid the extra object creation, yeah. it also makes lifetime of `attribution` clearer. just need to doc that it's undefined for local-only segments.